src/player.cpp

```diff
@@ -4678,15 +4678,22 @@ bool player::is_dead_state() const
 
 void player::on_dodge( Creature *source, int difficulty )
 {
-    if( difficulty == INT_MIN && source != nullptr ) {
-        difficulty = source->get_melee();
-    }
+    // Each avoided hit consumes an available dodge
+    // When no more available we are likely to fail player::dodge_roll
+    dodges_left--;
 
-    if( difficulty > 0 ) {
-        practice( skill_dodge, difficulty );
-    }
+    // Even if we are not to train still call practice to prevent skill rust
+    practice( skill_dodge, std::max( difficulty, 0 ) );
```

Comment:
Old version:
```practice( skill_dodge, source->get_melee() * 2, source->get_melee() );```

New version translates to:
```practice( skill_dodge, source->get_melee(), source->get_melee() );```

Because the `amount` is now equal to `cap`.